c++/CMakeLists.txt

```diff
@@ -51,6 +46,10 @@ if(CAPNP_LITE AND BUILD_TOOLS)
   message(WARNING "Command-line tools will not be built with CAPNP_LITE.")
 endif()
 
+if(BUILD_TOOLS AND EXTERNAL_CAPNP)
+  message(SEND_ERROR "EXTERNAL_CAPNP cannot be set with BUILD_TOOLS.")
```

Comment:
> compiler flags in your project that don't work with Cap'n Proto

My understanding is that cmake propagates flags only down the directory hierarchy. If directories above `capnp` don't set them compiler flags, no harm to `capnp` can happen.

> global targets get conflated

This can be seen also as a feature. E.g. `make test` runs also capnp tests. That is desired behavior.

I understand that this is not good schema for packaging. But it is very convenient for parallel development on capnp and main project. Thank you for your feedback!